File: server/java/metadata/src/main/java/com/metaobjects/agentcontext/AssembledFile.java
package com.metaobjects.agentcontext;

/**
 * A file the assembler emits, {@code path} relative to the consumer project root.
 *
 * @param path     consumer-relative output path (forward-slash separated).
 * @param contents the file contents (UTF-8 string; byte-identical to the reference).
 */
public record AssembledFile(String path, String contents) {
}

File: server/csharp/MetaObjects/AgentContext/AssembledFile.cs
// AssembledFile — a (path, contents) pair the assembler emits, path relative to
// the consumer project root.

namespace MetaObjects.AgentContext;

/// <summary>A file the assembler emits, <c>Path</c> relative to the consumer project root.</summary>
/// <param name="Path">Consumer-relative output path (forward-slash separated).</param>
/// <param name="Contents">The exact file contents (byte-faithful UTF-8 string).</param>
public sealed record AssembledFile(string Path, string Contents);
